### PR TITLE
Add support for Python 3.10

### LINKED ISSUE #615: Add support for Python 3.10
I see IDAES now tests against Python 3.10. We should do the same, and if for some reason we have an incompatibility we should address it.

### PR DESCRIPTION
**Pull request by lbianchi-lbl**
## Summary/Motivation:

- It would be nice

## Changes proposed in this PR:

- Add 3.10 to Python version specs in `setup.py`
- Add 3.10 to CI workflow job matrix

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -97,7 +98,7 @@
         "fastjsonschema",  # schema validation
         "click",  # command-line tools with Click
         # tutorial tests
-        "nbformat",
+        # "nbformat",
```
Comment (lbianchi-lbl) [inline setup.py]:
```suggestion
        "nbformat",
```

File: setup.py [hunk 2]
```diff
@@ -86,7 +87,7 @@
     ],
     keywords="water systems, chemical engineering, process modeling, filtration, desalination, nawi",
     packages=find_namespace_packages(),
-    python_requires=">=3.7, <4",
+    python_requires=">=3.7,<3.11",
```
Comment (bknueven) [inline setup.py]:
```suggestion
    python_requires=">=3.7",
```
Is the `<3.11` worth it? Before this PR WaterTAP would happily install on 3.10...

Comment (bknueven) [inline setup.py]:
```suggestion
    python_requires=">=3.7,<3.99",
```
An alternative ...

**Review by bknueven — APPROVED** [review]
Happy with this as-is -- just a minor question over compatibility with unreleased Python versions.

**Pull request merged**